Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -91,32 +101,74 @@ void HiresTexture::Update()
 
   const std::string& game_id = SConfig::GetInstance().GetGameID();
   const std::string texture_directory = GetTextureDirectory(game_id);
-  std::vector<std::string> extensions{
-      ".png", ".bmp", ".tga", ".dds",
-      ".jpg"  // Why not? Could be useful for large photo-like textures
-  };
+  std::vector<std::string> extensions{".dds", ".png", ".bmp",  ".tga",  ".tgb",
+                                      ".jpg", ".jpe", ".jpeg", ".jfif", ".hdr"};
 
-  std::vector<std::string> filenames =
+  std::vector<std::string> Filepaths_Found =
       Common::DoFileSearch({texture_directory}, extensions, /*recursive*/ true);
 
   const std::string code = game_id + "_";
 
-  for (auto& rFilename : filenames)
+  FileType_Counter_DDS_Amount = 0;
+  FileType_Counter_PNG_Amount = 0;
+  FileType_Counter_BMP_Amount = 0;
+  FileType_Counter_TGA_Amount = 0;  // Common for TGA and TGB vatiants
+  FileType_Counter_JPG_Amount = 0;  // Common for JPG, JPE, JPEG and JFIF variants
+  FileType_Counter_HDR_Amount = 0;
+  FileType_Counter_Total_Amount = 0;
+
+  for (auto& rFilepath : Filepaths_Found)
   {
-    std::string FileName;
-    SplitPath(rFilename, nullptr, &FileName, nullptr);
+    std::string CurrentFilename;
+    SplitPath(rFilepath, nullptr, &CurrentFilename, nullptr);
+    SplitPath(rFilepath, nullptr, nullptr, &FileType_Counter_CurrentType);
 
-    if (FileName.substr(0, code.length()) == code)
+    if (CurrentFilename.substr(0, code.length()) == code)
     {
-      s_textureMap[FileName] = rFilename;
+      s_textureMap[CurrentFilename] = rFilepath;
       s_check_native_format = true;
     }
 
-    if (FileName.substr(0, s_format_prefix.length()) == s_format_prefix)
+    if (CurrentFilename.substr(0, s_format_prefix.length()) == s_format_prefix)
     {
-      s_textureMap[FileName] = rFilename;
+      s_textureMap[CurrentFilename] = rFilepath;
       s_check_new_format = true;
     }
+
+    // Check FileType And Count - used by prefetcher
+    if ((FileType_Counter_CurrentType == ".dds") || (FileType_Counter_CurrentType == ".DDS"))
```

Comment:
perhaps use `std::transform(FileType_Counter_CurrentType.begin(), FileType_Counter_CurrentType.end(), FileType_Counter_CurrentType.begin(), ::tolower);` before against checking to cut half the cases? This also catches the cases where the filetype is malformed (ie. `.jPg`).